libmscore/score.cpp

```diff
@@ -2664,6 +2665,19 @@ void Score::cmdConcertPitchChanged(bool flag, bool /*useDoubleSharpsFlats*/)
 
             transposeKeys(staffIdx, staffIdx + 1, Fraction(0,1), lastSegment()->tick(), interval, true, !flag);
 
+            for (Segment* segment = firstSegment(SegmentType::Ambitus); segment; segment = segment->next1(SegmentType::Ambitus)) {
+                  //interval = staff->part()->instrument(segment->tick())->transpose();
+                  //if (!flag)
+                        //interval.flip();
+                  for (Element* e : segment->elist()) {
+                        if (e->isAmbitus()) {
+                              Ambitus* a = toAmbitus(e);
+                              a->updateRange();
```

Comment:
This is the wrong approach.

Ambiti may be hand-edited in inspector (which I often do if there’s an optional extra note that can be left out that is rather out of a normal range). Toggling concert pitch should **not** update the range from the score; it must rather transpose the top and bottom notes the same as it does to regular notes in that stave.

Reply:
Hmm, I see. But a) how to and b) how to get it to work for octave transposing instruments too?

Reply:
The problem is in the <code>break</code> **after** the <code>a->updateRange();</code>. Now the iterations stops at the first <code>Ambitus</code> found but the segment might contain more <code>Ambitus</code>'s.

Reply:
Segment, of course, more Ambiti in the same measure, but different staves! Bad thinko on my side I guess.

Reply:
I think it's better the move the <code>Ambitus</code> loop outside the <code>for (Staff* staff : _staves)</code> loop since you end up doing the same thing <code>nstaves()</code> times.
This solves your first problem but still leaves the issue as mentioned by @mirabilos . I indeed do similar things, especially when newer instruments have a larger range (e.g. Baritone Saxophones, even Bb Clarinets which can be extended in the lower range)

Reply:
Still needs the check for whether the staff is transposing or not, at least as an optimization. Would solve (a good part of) @mirabilos' issue too, as there it is not about transposing instruments but human singing voices ;-)

Reply:
@mirabilos you can avoid the manual changes of the Ambitus by disabling "Play" for those extra notes.

Reply:
Hmm, bad, that change to skip non-transposing instruments/segments causes a crash :-(